### ISSUE TITLE
tcsetattr(STDIN_FILENO, TCSAFLUSH, &raw) hangs up tty

### ISSUE THREAD

**Issue by Rhialto** [UnNetHack/UnNetHack#47]
At https://github.com/UnNetHack/UnNetHack/blob/master/sys/share/unixtty.c#L469 there is an attempt to set the terminal to raw mode. Unfortunately it also seems to set the speed to 0. This has the effect of terminating the login session and even the xterm (if that's what you're using) to close.

This patch avoids the issue:
```--- sys/share/unixtty.c.orig    2022-05-11 18:37:13.878580520 +0000
+++ sys/share/unixtty.c
@@ -462,7 +462,8 @@ check_utf8_console()
     }
 
     /* set minimal raw mode */
-    struct termios raw = { 0 };
+    struct termios raw = original;
+    cfmakeraw(&raw);
     raw.c_cc[VMIN] = 0;
     raw.c_cc[VTIME] = 3; /* timeout of 0.3 seconds */
     if (tcsetattr(STDIN_FILENO, TCSAFLUSH, &raw) == -1) {

```
My manual tcsetattr(3) says:
```
     The cfmakeraw function sets the flags stored in the termios structure
     (initialized by tcgetattr) to a state disabling all input and output
     processing, giving a "raw I/O path".  It should be noted that there is no
     function to reverse this effect.  This is because there are a variety of
     processing options that could be re-enabled and the correct method is for
     an application to snapshot the current terminal state using the function
     tcgetattr, setting raw mode with cfmakeraw and the subsequent tcsetattr,
     and then using another tcsetattr with the saved state to revert to the
     previous terminal state.
```

**Comment by bhaak** [comment]
I can't reproduce the closing of xterm with version 372 on a Linux Debian 12.

The code you are proposing seems reasonable though and I couldn't find any detrimental effects of doing it this way.

**Comment by Rhialto** [comment]
I guess is is a difference between Linux and *BSD. Setting the baud rate to 0 hangs up the terminal. I tried `stty 0` on a linux vm and I got the error message `stty: 'standard input': unable to perform all requested operations`,
On NetBSD however (and I suspect on other BSDs too), `stty 0` actually does what you tell it to do, and the terminal session ends. I just can't find any more where it's documented, but it has been like that forever.
